File: src/main/java/com/chii/homemanagement/entity/Reminder.java
package com.chii.homemanagement.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 提醒实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("reminder")
@Schema(description = "提醒信息")
public class Reminder {

    @TableId(type = IdType.AUTO)
    @Schema(description = "主键ID")
    private Long id;

    /**
     * 提醒类型: warranty-保修到期，maintenance-定期维护，lending-借用归还，expiry-物品过期，other-其他
     */
    @Schema(description = "提醒类型", allowableValues = {"warranty", "maintenance", "lending", "expiry", "other"}, required = true)
    private String type;

    /**
     * 关联实体ID
     */
    @Schema(description = "关联实体ID", required = true)
    private Long entityId;

    /**
     * 提醒标题
     */
    @Schema(description = "提醒标题", required = true, example = "保修到期提醒")
    private String title;

    /**
     * 提醒内容
     */
    @Schema(description = "提醒内容", example = "您的物品保修即将到期，请及时处理。")
    private String content;

    /**
     * 提醒日期
     */
    @Schema(description = "提醒日期", required = true, example = "2023-01-01")
    private LocalDate remindDate;

    /**
     * 状态: pending-待提醒，sent-已提醒，processed-已处理，ignored-已忽略
     */
    @Schema(description = "提醒状态", allowableValues = {"pending", "sent", "processed", "ignored"}, defaultValue = "pending")
    private String status;

    /**
     * 所属用户ID
     */
    @Schema(description = "所属用户ID", required = true)
    private Long userId;
    
    /**
     * 创建用户ID
     */
    @Schema(description = "创建用户ID")
    private Long createUserId;

    /**
     * 创建时间
     */
    @Schema(description = "创建时间", example = "2023-01-01 12:00:00")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Schema(description = "更新时间", example = "2023-01-01 12:00:00")
    private LocalDateTime updateTime;
}

File: vue3/src/types/reminder.ts
// import type { PageParams } from "@/types/entity";

/**
 * 提醒类型
 */
export type ReminderType = "warranty" | "maintenance" | "lending" | "expiry" | "other";

/**
 * 提醒状态
 */
export type ReminderStatus = "pending" | "sent" | "processed" | "ignored";

/**
 * 通知方式
 */
export type NotificationMethod = "system" | "email" | "sms" | "wx";

/**
 * 重复周期
 */
export type RecurringCycle = "daily" | "weekly" | "monthly" | "yearly";

/**
 * 提醒实体
 */
export interface Reminder {
  /**
   * 提醒ID
   */
  id: string;
  /**
   * 物品ID
   */
  entityId: string;
  /**
   * 物品名称
   */
  entityName?: string;
  /**
   * 用户ID
   */
  userId: string;
  /**
   * 提醒类型
   */
  type: ReminderType;
  /**
   * 提醒日期
   */
  remindDate: string;
  /**
   * 提醒状态
   */
  status: ReminderStatus;
  /**
   * 提醒内容
   */
  content: string;
  /**
   * 通知方式 (逗号分隔的字符串)
   */
  notificationMethods: string;
  /**
   * 提前提醒天数
   */
  daysInAdvance: number;
  /**
   * 是否重复提醒
   */
  isRecurring: boolean;
  /**
   * 重复周期
   */
  recurringCycle?: RecurringCycle;
  /**
   * 创建者ID
   */
  createUserId: number;
  /**
   * 创建时间
   */
  createTime: string;
  /**
   * 更新时间
   */
  updateTime: string;
}

/**
 * 提醒查询参数
 */
export interface ReminderQueryParams {
  /**
   * 物品名称
   */
  entityId?: number;
    /**
   * 物品名称
   */
  entityName: string;
  /**
   * 提醒类型
   */
  type?: ReminderType;
  /**
   * 提醒状态
   */
  status?: ReminderStatus;
  /**
   * 日期范围
   */
  dateRange?: [string, string];
  /**
   * 用户ID
   */
  userId?: number;
}

/**
 * 物品选项
 */
export interface ItemOption {
  id: number;
  name: string;
}
